src/hmr-server-engine.ts

```diff
@@ -71,31 +77,17 @@ export class EsmHmrEngine {
     entry.needsReplacement = state;
   }
 
-  broadcastMessage(channel: string, data: object) {
-    for (const client of this.clients) {
-      sendMessage(client, channel, JSON.stringify(data));
-    }
+  broadcastMessage(data: object) {
+    this.clients.forEach(client => {
+      client.send(JSON.stringify(data));
+    })
   }
 
-  connectClient(res: http.ServerResponse) {
-    res.writeHead(200, {
-      Connection: 'keep-alive',
-      'Content-Type': 'text/event-stream',
-      'Cache-Control': 'no-cache',
-      'Access-Control-Allow-Origin': '*',
-    });
-    sendMessage(res, 'connected', 'ready');
-    setInterval(sendMessage, 60000, res, 'ping', 'waiting');
-    this.clients.push(res);
+  connectClient(res) {
+    this.clients.add(res);
   }
 
-  disconnectClient(client: http.ServerResponse) {
```

Comment:
For all of these `connectClient`, `disconnectClient` we can probably now make them private, and manage all of this inside the engine itself without involving the server.